examples/todo.rs

```diff
@@ -30,16 +31,23 @@ use rustful::StatusCode;
 
 //Helper for setting a status code and then returning from a function
 macro_rules! or_abort {
-    ($e: expr, $response: expr, $status: expr) => (
+    ($e: expr, $response: ident, $error_message: ident) => (
         if let Some(v) = $e {
             v
         } else {
-            $response.set_status($status);
+            $response.set_status(StatusCode::BadRequest);
+            $response.headers_mut().set(ContentType(content_type!(Text / Plain; Charset = Utf8)));
+            $response.send($error_message);
             return
         }
     )
 }
 
+type ErrorMessage = &'static str;
```

Comment:
Yeah, honestly I don't know how to do this in Rust. All I want is declare some enumeration of all errors this example could produce. And then explicitly require this type to be fed to macro rule.